src/commands/lobby.ts

```diff
@@ -1,50 +1,75 @@
-        const actionRow = new ActionRowBuilder<ButtonBuilder>().addComponents(
-            new ButtonBuilder()
-                .setCustomId("join-game")
-                .setLabel("Join")
-                .setStyle(ButtonStyle.Success),
-
-            new ButtonBuilder()
-                .setCustomId("leave-game")
-                .setLabel("Leave")
-                .setStyle(ButtonStyle.Danger),
-
-            new ButtonBuilder()
-                .setCustomId("start-game")
-                .setLabel("Start Now")
-                .setStyle(ButtonStyle.Primary)
-        );
-
-        await interaction.reply({
+        console.log("A lobby has been created");
+
+        // create and display the lobby emebed
+        const { embeds: embed, components: actionRow } =
+            lobbyEmbed.CreateEmbed();
+
+        const message = await interaction.reply({
             embeds: [embed],
             components: [actionRow],
         });
 
+        const collector = message.createMessageComponentCollector({
+            filter: (i) => i.user.id === interaction.user.id,
+            componentType: ComponentType.Button,
+            time: 120_000,
+        });
+
+        // collector that responds to the buttons
+        collector.on("collect", async (i) => {
+            if (i.customId === "join-game") {
+                // add player if not already in the lobby
+                if (!gameManager.players.getPlayer(interaction.member.id)) {
+                    gameManager.players.addPlayer(
+                        await fetchMember(
+                            interaction.guildId,
+                            interaction.member.id
+                        ),
+                        interaction.member
+                    );
+                    // update lobby player list
+                    const { embeds: updatePlayers } = lobbyEmbed.CreateEmbed();
+                    interaction.editReply({ embeds: [updatePlayers] });
+                }
+                console.log(`Added player ${interaction.member.displayName}`);
+                await i.deferUpdate();
+            }
+
+            if (i.customId === "leave-game") {
+                // remove player if in the lobby
+                if (gameManager.players.getPlayer(interaction.member.id)) {
+                    gameManager.players.removePlayer(interaction.member.id);
+                    // update lobby player list
+                    const { embeds: updatePlayers } = lobbyEmbed.CreateEmbed();
+                    interaction.editReply({ embeds: [updatePlayers] });
+                }
+                console.log(`Removed player ${interaction.member.displayName}`);
+                await i.deferUpdate();
+            }
+
+            if (i.customId === "start-game") {
+                gameManager.start();
+                await i.deferUpdate();
+            }
+        });
+
         // starts game in two minutes
         setTimeout(async () => {
```

Comment:
Check to see that the game manager's `.currentState` is still "waiting".
This prevents 2 calls to `.start()` if the host decides to start the game early with the "start" button.